### DISCUSSION TITLE
Type safe routes and params

### DISCUSSION THREAD

**Discussion by chrisdrackett** [nandorojo/solito#387] [General]
I'm excited to find this library as we've been doing this internally for a couple years and now with a couple small code changes I won't have to maintain it by myself anymore 😓 😉

One question I have is around making the use of routes and params type safe and if this project has any suggestions about ways to do this.

At the moment we handle this by defining the types for our routes and params by hand and then using this with custom components/hooks that can then validate if a route exists and if it has the correct number and type of param. Obviously this isn't very dry as it requires an update to both the types and the file structure if a route changes. This generally works out, especially as a project ships as we don't generally ever want to touch an existing href that is live, but I still thought it was worth asking if there is a recommended pattern for this in Solito.

**Comment by nandorojo** [comment]
hey! so i currently achieve this with pathpida. i make the routes in next.js pages folder, export a `type Query` in each page, and then in react navigation’s params / useParams cases, I infer query params based on the Pathpida `query` variable.

maybe i can put together some snippets at some point

**Reply by shoelessness** [reply]
@nandorojo have you had a chance to try out https://docs.expo.dev/router/reference/typed-routes/ ? Looks like there could be potential there for something that's less manual than Pathipida.
